src/semantic-highlighting.ts

```diff
@@ -2,14 +2,15 @@ import * as fs from 'fs';
 import * as jsonc from 'jsonc-parser';
 import * as path from 'path';
 import * as vscode from 'vscode';
-import * as vscodelc from 'vscode-languageclient';
+import * as vscodelc from 'vscode-languageclient/node';
 import * as vscodelct from 'vscode-languageserver-types';
 
 export function activate(client: vscodelc.LanguageClient,
                          context: vscode.ExtensionContext) {
   const feature = new SemanticHighlightingFeature(client, context);
   context.subscriptions.push(feature);
   client.registerFeature(feature);
+  client.registerProposedFeatures(); // enables LSP 3.16 semantic highlighting
```

Comment:
I think this should be moved to `extension.ts`. This file is the legacy implementation, and is enabled by the "clangd.semanticHighlighting" flag.

Reply:
Ah so it should always be enabled?

Reply:
not really, it will be controlled by the vscode's builtin "Semantich Highlighting" flag (`"editor.semanticHighlighting.enabled"`)